voidcrew/code/modules/overmap/dynamic_object_defines.dm

```diff
@@ -14,6 +14,9 @@
 	var/preserve_level = FALSE
 	///What kind of planet the level is, if it's a planet at all.
 	var/datum/overmap/planet/planet
+	///Keep track of whether or not the docks have been reserved by a ship. This is required to prevent issues where two ships will attempt to dock in the same place due to unfortunate timing
+	var/first_dock_taken = FALSE
+	var/second_dock_taken = FALSE
```

Comment:
couldn't this just be done as an int? and then you check like `if (used_docks >= 2) ... dont dock`

Reply:
Thats going to make it a lot harder to read, since the second dock could be taken but not the first. Its doable but im gonna need to define a bunch of ints and I think if we ever want to add more than 2 docks in the future then that approach is gonna make it even more confusing